From 159122e835c80c35c10b300e3bd19ecaa5343f7d Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Thu, 4 Oct 2018 12:15:24 +0300
Subject: [PATCH 0711/1051] arm64: dts: fix armada-37xx configuration

armada-3720-db.dts - describe basic configuration with
eth0 (rgmii) port configuration
armada-3720-db-B.dts - describe two port configuration:
eth0 (rgmii) and eth1 (sgmii 1G)

Change-Id: I483636b276b81bef3d63f0f8d91d115a546497d1
Signed-off-by: Yelena Krivosheev <yelena@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60227
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db-B.dts |  9 +++++++++
 arch/arm64/boot/dts/marvell/armada-3720-db.dts   | 12 +-----------
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
index bcfae3f2054d..8c1bdd6bdb0b 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
@@ -12,4 +12,13 @@
 	status = "okay";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&a3700_comphy0 1>;
+	phy-mode = "sgmii";
+	phy = <&phy1>;
+};
+
+&mdio {
+	status = "okay";
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index a0566d71af06..e8cbc1fa0d07 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -78,13 +78,6 @@
 	status = "okay";
 };
 
-/* Gigabit module on CON18(V2.0)/CON20(V1.4) */
-&eth1 {
-	phy-mode = "sgmii";
-	phy = <&phy1>;
-	status = "okay";
-};
-
 &i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1_pins>;
@@ -115,15 +108,12 @@
 	};
 };
 
+/* Gigabit module on CON19(V2.0)/CON21(V1.4) only */
 &mdio {
 	status = "okay";
 	phy0: ethernet-phy@0 {
 		reg = <0>;
 	};
-
-	phy1: ethernet-phy@1 {
-		reg = <1>;
-	};
 };
 
 /* CON15(V2.0)/CON17(V1.4) : PCIe / CON15(V2.0)/CON12(V1.4) :mini-PCIe */
-- 
2.17.1

